adapters/repos/db/vector/hnsw/vector_cache.go

```diff
@@ -41,19 +42,29 @@ type shardedLockCache struct {
 
 var shardFactor = uint64(512)
 
+const defaultDeletionInterval = 3 * time.Second
+
 func newShardedLockCache(vecForID VectorForID, maxSize int,
 	logger logrus.FieldLogger, normalizeOnRead bool,
+) *shardedLockCache {
+	return newShardedLockCacheInternal(vecForID, maxSize, logger,
+		normalizeOnRead, defaultDeletionInterval)
+}
+
+func newShardedLockCacheInternal(vecForID VectorForID, maxSize int,
```

Comment:
I think the previous implementation is better for couple of reasons:
1. `newShardedLockCacheInternal()` is called once by `newShardedLockCache()`.
2. Both functions are private functions.
3. Unlike the previous implementation this one introduces a new function call overhead.
4. In terms of unit testing the previous implementation has the same benefits and in addition it has less code.
5. IMHO this technique would make more sense if `newShardedLockCache` was public

In consequence I would remove `newShardedLockCacheInternal()` and delegate its duties to `newShardedLockCache()`

Reply:
true. nicely explained 👍 . I will revert recent changes.